Run started:2024-11-28 11:40:45.103250

Test results:
>> Issue: [B105:hardcoded_password_string] Possible hardcoded password: 'your_secret_key'
   Severity: Low   Confidence: Medium
   CWE: CWE-259 (https://cwe.mitre.org/data/definitions/259.html)
   More Info: https://bandit.readthedocs.io/en/1.8.0/plugins/b105_hardcoded_password_string.html
   Location: ./authentication/app.py:15:11
14	app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
15	app.config['SECRET_KEY'] = 'your_secret_key'
16	 

--------------------------------------------------
>> Issue: [B113:request_without_timeout] Call to requests without timeout
   Severity: Medium   Confidence: Low
   CWE: CWE-400 (https://cwe.mitre.org/data/definitions/400.html)
   More Info: https://bandit.readthedocs.io/en/1.8.0/plugins/b113_request_without_timeout.html
   Location: ./authentication/app.py:118:25
117	        headers = {'x-access-token': token}  # Aggiungi il token all'header
118	        gacha_response = requests.delete(f"{GACHA_SERVICE_URL}/gacha_service/players/{account_id}/gachas", headers=headers)
119	 

--------------------------------------------------
>> Issue: [B201:flask_debug_true] A Flask app appears to be run with debug=True, which exposes the Werkzeug debugger and allows the execution of arbitrary code.
   Severity: High   Confidence: Medium
   CWE: CWE-94 (https://cwe.mitre.org/data/definitions/94.html)
   More Info: https://bandit.readthedocs.io/en/1.8.0/plugins/b201_flask_debug_true.html
   Location: ./authentication/app.py:315:4
314	if __name__ == '__main__':
315	    app.run(debug=True)

--------------------------------------------------
>> Issue: [B105:hardcoded_password_string] Possible hardcoded password: 'your_secret_key'
   Severity: Low   Confidence: Medium
   CWE: CWE-259 (https://cwe.mitre.org/data/definitions/259.html)
   More Info: https://bandit.readthedocs.io/en/1.8.0/plugins/b105_hardcoded_password_string.html
   Location: ./authentication/old-app.py:21:11
20	app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
21	app.config['SECRET_KEY'] = 'your_secret_key'
22	

--------------------------------------------------
>> Issue: [B113:request_without_timeout] Call to requests without timeout
   Severity: Medium   Confidence: Low
   CWE: CWE-400 (https://cwe.mitre.org/data/definitions/400.html)
   More Info: https://bandit.readthedocs.io/en/1.8.0/plugins/b113_request_without_timeout.html
   Location: ./authentication/old-app.py:136:25
135	        # Effettuare una richiesta DELETE al servizio gacha_service per eliminare la gacha collection dell'utente
136	        gacha_response = requests.delete(f"{GACHA_SERVICE_URL}/gacha_service/players/{account_id}/gachas")
137	

--------------------------------------------------
>> Issue: [B201:flask_debug_true] A Flask app appears to be run with debug=True, which exposes the Werkzeug debugger and allows the execution of arbitrary code.
   Severity: High   Confidence: Medium
   CWE: CWE-94 (https://cwe.mitre.org/data/definitions/94.html)
   More Info: https://bandit.readthedocs.io/en/1.8.0/plugins/b201_flask_debug_true.html
   Location: ./authentication/old-app.py:369:4
368	if __name__ == 'main':
369	    app.run(debug=True)

--------------------------------------------------
>> Issue: [B105:hardcoded_password_string] Possible hardcoded password: 'your_secret_key'
   Severity: Low   Confidence: Medium
   CWE: CWE-259 (https://cwe.mitre.org/data/definitions/259.html)
   More Info: https://bandit.readthedocs.io/en/1.8.0/plugins/b105_hardcoded_password_string.html
   Location: ./authentication/origin-app.py:18:11
17	app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
18	app.config['SECRET_KEY'] = 'your_secret_key'
19	

--------------------------------------------------
>> Issue: [B113:request_without_timeout] Call to requests without timeout
   Severity: Medium   Confidence: Low
   CWE: CWE-400 (https://cwe.mitre.org/data/definitions/400.html)
   More Info: https://bandit.readthedocs.io/en/1.8.0/plugins/b113_request_without_timeout.html
   Location: ./authentication/origin-app.py:103:25
102	        # Effettuare una richiesta DELETE al servizio gacha_service per eliminare la gacha collection dell'utente
103	        gacha_response = requests.delete(f"{GACHA_SERVICE_URL}/gacha_service/players/{account_id}/gachas")
104	

--------------------------------------------------
>> Issue: [B201:flask_debug_true] A Flask app appears to be run with debug=True, which exposes the Werkzeug debugger and allows the execution of arbitrary code.
   Severity: High   Confidence: Medium
   CWE: CWE-94 (https://cwe.mitre.org/data/definitions/94.html)
   More Info: https://bandit.readthedocs.io/en/1.8.0/plugins/b201_flask_debug_true.html
   Location: ./authentication/origin-app.py:330:4
329	if __name__ == 'main':
330	    app.run(debug=True)

--------------------------------------------------

Code scanned:
	Total lines of code: 639
	Total lines skipped (#nosec): 0
	Total potential issues skipped due to specifically being disabled (e.g., #nosec BXXX): 0

Run metrics:
	Total issues (by severity):
		Undefined: 0
		Low: 3
		Medium: 3
		High: 3
	Total issues (by confidence):
		Undefined: 0
		Low: 3
		Medium: 6
		High: 0
Files skipped (0):
